docs/spec/spec.md

```diff
@@ -0,0 +1,950 @@
+
+# 3. Protocol Buffers to Ballerina Data Mapping
+
+The following table illustrates the data mapping of protocol buffers data types to relevant Ballerina types.
+
+|Protobuf Type|Ballerina Type|
+|---|---|
+|google.protobuf.DoubleValue|float|
+|google.protobuf.FloatValue|float|
+|google.protobuf.Int64Value|int|
+|google.protobuf.UInt64Value|int|
+|google.protobuf.Int32Value|int|
+|google.protobuf.UInt32Value|int|
+|google.protobuf.BoolValue|boolean|
+|google.protobuf.StringValue|string|
+|google.protobuf.BytesValue|byte[]|
+|google.protobuf.Empty|()|
+|google.protobuf.Timestamp|time:Utc|
+|google.protobuf.Duration|time:Seconds|
+|google.protobuf.Any|'any:Any|
+
+Note that here the `'any` is the namespace of the `ballerina/protobuf.types.'any` submodule. Additionally, the `google.protobuf.Any` need serialization and deserialization mechanisms. To do that, `ballerina/protobuf.types.'any` module contains two APIs called pack and unpack to serialize and deserialize `Any` type records.
+
+```ballerina
+# Generate and return the generic `'any:Any` record that is used to represent protobuf `Any` type.
+#
+# + message - The record or the scalar value to be packed as Any type
+# + return - Any value representation of the given message
+public isolated function pack(ValueType message) returns Any;
+
+# Unpack and return the specified Ballerina value
+#
+# + anyValue - Any value to be unpacked
+# + targetTypeOfAny - Type descriptor of the return value
+# + return - Return a value of the given type
+public isolated function unpack(Any anyValue, ValueTypeDesc targetTypeOfAny = <>) returns targetTypeOfAny|Error;
+```
+
+# 4. gRPC Communication
+
+gRPC has 4 types of RPCs (Remote Procedure Calls), and Ballerina supports all of them.
+1. Unary
+2. Server streaming
+3. Client streaming
+4. Bidirectional streaming
+
+Note that, to explain the behaviour of these 4 RPC types, this document uses the standard Route Guide example.
+- [Details of the route guide example](https://grpc.io/docs/languages/go/basics/)
+- [Protocol buffer definition of the route guide example](https://github.com/ballerina-platform/module-ballerina-grpc/blob/674bda12a90f99c2735badc5567cd7dd7e14ba09/examples/routeguide/proto-file/route_guide.proto)
+
+## 4.1. Unary RPC
+
+The RPC service definition of a unary call is as follows.
+```proto
+service RouteGuide {
+    rpc GetFeature(Point) returns (Feature) {}
```

Comment:
```suggestion
    rpc GetFeature(Point) returns (Feature) {};
```

Reply:
Semicolon is not needed https://github.com/grpc/grpc-go/blob/v1.38.0/examples/route_guide/routeguide/route_guide.proto